src/data.py

```diff
@@ -227,6 +228,107 @@ def __call__(self, sample):
         return (sample - sample.mean()) / (sample.std() + 1e-8)
 
 
+class SlidingWindowNormalize:
+    """ Based on Sliding window augmentations of
+    https://github.com/cchinchristopherj/Right-Whale-Convolutional-Neural-Network/blob/master/whale_cnn.py
+        Translated to torch
+        Has 50/50 chance of activating H sliding window or V sliding window
+
+        Must come after spectrogram and AmplitudeToDB
+    """
+
+    def __init__(self, sr: float, n_fft: int, lower_cutoff: float = 50, norm=True,
+                 inner_ratio: float = 0.06, outer_ratio: float = 0.5):
+        self.sr = sr
+        self.n_fft = n_fft
+        self.lower_cutoff = lower_cutoff
+        self.norm = norm
+        self.inner_ratio = inner_ratio
+        self.outer_ratio = outer_ratio
+
+    # slidingWindowV Function from: https://github.com/nmkridler/moby2/blob/master/metrics.py
+    def slidingWindowV(self, P):
+        ''' slidingWindowV Method
+                Enhance the contrast vertically (along frequency dimension)
+
+                Args:
+                    P: 2-D numpy array image
+                    inner: int length of inner exclusion region (to calculate local mean)
+                    outer: int length of the window (to calculate overall mean)
+                    maxM: int size of the output image in the y-dimension
+                    norm: boolean indicating whether or not to cut off extreme values
+                Returns:
+                    Q: 2-D numpy array image with vertically-enhanced contrast
+
+        '''
+        min_f_ind = int((self.lower_cutoff / (self.sr / 2)) * self.n_fft)
+        Q = P.cpu().clone().numpy()
```

Comment:
can we please change Q and P to q and p? or even better, let's give them indicative names :)